pynamodb/indexes.py

```diff
@@ -22,15 +23,15 @@
 _M = TypeVar('_M', bound='Model')
 
 
-class IndexMeta(type(Generic)):
+class IndexMeta(GenericMeta):
     """
     Index meta class
 
     This class is here to allow for an index `Meta` class
     that contains the index settings
     """
     def __init__(cls, name, bases, attrs, *args, **kwargs):
-        super().__init__(name, bases, attrs, *args, **kwargs)
+        super().__init__(name, bases, attrs, *args, **kwargs)  # type: ignore
```

Comment:
No way to make mypy happier here: its definition of GenericMeta is incorrect as it doesn't accept `tvars` etc.